ui-components/src/blocks/Card.docs.mdx

```diff
@@ -0,0 +1,60 @@
+import { Canvas, Story, ArgsTable } from "@storybook/addon-docs"
+import { Swatch } from "../prototypes/Swatch"
+import { Card } from "./Card"
+
+# Card
+
+<Canvas>
+  <Story id="blocks-card--text-content" />
+</Canvas>
+
+The 2nd-generation Card component is intended to serve as a flexible content container with a header, optional content sections, and a footer. Over time our intention is to "port" some other card types and content blocks in the system to use Card as their underlying structure.
```

Comment:
I'm finding these docs / the way this component has been set up to be a bit confusing to me. We aren't using this kind of subcomponent structure anywhere else vs just exporting the pieces separately - is there a benefit outside of style to do it this way?

Reply:
Sure, so what I'm proposing is a bit of a stylistic shift away from supplying "subcomponent" content as extra props on a top-level component (e.g. Card), and instead using child components for those pieces. So instead of something like:

```jsx
<Card header={<HeadingGroup title="Title" subtitle="Subtitle" />} footer={<ButtonGroup ... />}>
```
etc.

We can do this:

```jsx
<Card>
  <Card.Header>
    <HeadingGroup title="Title" subtitle="Subtitle" />
  </Card.Header>
  <Card.Footer>
    <ButtonGroup ... />
  </Card.Footer>
</Card>
```
etc.

The idea being that `Card` is both a top-level component and a namespace: Card (namespace) .Header (subcomponent)

The additional code in the prototypes folder was an example of pushing this idea even further so that subcomponents could be "slotted" into specific places within the DOM structure of the top-level component. In this case we're sticking with more of a simplistic structure where all the children of Card are fine with rendering in order from top to bottom within the parent.

Exporting the pieces separately (as `CardHeader`, `CardFooter`) is done so simply for the ergonomics of supplying those "atoms" separately to a developer who wants to create a different component that is Card-like. Normally a consumer wouldn't need to use those exports standalone, as generally they don't provide much utility on their own.

I was inspired on some of these aspects by looking at other component libraries…for example Semantic UI. Here's their example of a Modal: https://react.semantic-ui.com/modules/modal/#types-modal  You can see they offer `<Modal.Header>`, `<Modal.Actions>`, etc. Contrast that with our modal which currently has a `title` prop, an `actions` prop, etc.

Functionality I suppose we're not talking about much of a difference at the end of the day, but personally I have found the use of React props for supplying lots of content and subcomponents to be rather cumbersome, and I appreciate the use of slots in other frameworks like Vue or in web components, and this sort of gets us closer to that style.

Certainly open to feedback on how all this might play into our increased focus on staying away from supplying large monolithic components.

Reply:
I didn't realize that this would be the first component to switch to this sub component model. I'm a big fan of the way Jared is doing it here. Having used semantic-ui and several other component libraries that use this pattern I personally like this way better than having each section be a prop to the main component.

However, there are a few things that we might want to discuss as a larger team. The biggest being how strict do we want the component to be. Like do we want our consumers to be able to put the footer before the header? Or do we allow any immediate children of the Card be anything besides `Card.Header`, `Card.Section`, or `Card.Footer`. The less strict we are the more likely a bug will appear, but the more strict means potentially more requests from new use cases or consumers.

The way it is being done in the prototype is a good way to add more restrictions on how the card is used. For example a user can no longer put the footer before the header. And I like where that component is heading.

But for this story I'd recommend not going down the path of "slotting" things in and keeping with the simplicity of the Card.tsx. 

The only call out I'm a bit worried about in this doc is encouraging creating a new type of Card but still adding the `.card` class. I'm not a fan of using a class built for a component outside of the component

Reply:
@ludtkemorgan Thanks for your feedback. I do like the thought of being more strict about how stuff can get slotted in like the prototype shows, but I feel like we'd only want to adapt that pattern if we think we'll use it in other places and not just the Card component.

Regarding the `.card` class, at one point I had tried something along the lines of splitting out all the CSS variables like `--background-color` from the class itself…I forget what I called it but let's say it was `.card-variables`. Then the Card component itself would use both `.card` and `.card-variables`, but some other component could opt to use only `.card-variables`. But then I felt like maybe that was overly complicating matters, and it's another pattern we haven't used anywhere else yet. Curious what you think about that.